src/software/utils/main_imageProcessing.cpp

```diff
@@ -120,128 +339,210 @@ int aliceVision_main(int argc, char * argv[])
       // Set verbose level
     system::Logger::get()->setLogLevel(verboseLevel);
 
-    if (downscale < 0.0001f || downscale > 1.0f)
+    // check user choose at least one input option
+    if(inputExpression.empty() && inputFolders.empty())
     {
-        ALICEVISION_LOG_ERROR("Invalid scale factor, downscale should be in range [0.0001, 1].");
+        ALICEVISION_LOG_ERROR("Program need at least --input or --inputFolders option." << std::endl << "No input images here.");
         return EXIT_FAILURE;
     }
 
-      // Analyze output path
-    const boost::filesystem::path path(sfmOutputDataFilepath);
-    const std::string outputPath = path.parent_path().string();
+#if !ALICEVISION_IS_DEFINED(ALICEVISION_HAVE_OPENCV)
+    if(pParams.bilateralFilter || pParams.ClaheFilter)
+    {
+        ALICEVISION_LOG_ERROR("Invalid option: BilateralFilter and ClaheFilter can't be used without openCV !");
+        return EXIT_FAILURE;
+    }
+#endif
 
-      // Read input
-    sfmData::SfMData sfmData;
-    if(!sfmDataIO::Load(sfmData, sfmInputDataFilename, sfmDataIO::ESfMData(sfmDataIO::ALL)))
+    if (pParams.downscale < 0.0001f || pParams.downscale > 1.0f)
     {
-        ALICEVISION_LOG_ERROR("The input SfMData file '" << sfmInputDataFilename << "' cannot be read.");
+        ALICEVISION_LOG_ERROR("Invalid scale factor, downscale should be in range [0.0001, 1].");
         return EXIT_FAILURE;
     }
 
-    for (auto & viewIt : sfmData.getViews())
+    // Check if sfmInputDataFilename exist and is recognized as sfm data file
+    const std::string inputExt = boost::to_lower_copy(fs::path(inputExpression).extension().string());
+    static const std::array<std::string, 2> SFMSupportedExtensions = {".sfm", ".abc"};
```

Comment:
Start variables with lower case: `sfmSupportedExtensions`

Reply:
Fixed in https://github.com/alicevision/AliceVision/pull/806/commits/f7584167ca69f68bb430cb92284b5733df155fd7 .